framework/src/mesh/MooseMesh.C

```diff
@@ -1054,6 +1054,44 @@ MooseMesh::getBoundaryActiveNeighborElemIds(BoundaryID bid) const
   return neighbor_elems;
 }
 
+bool
+MooseMesh::isBoundaryInternalToSubdomains(BoundaryID bid,
+                                          const std::set<SubdomainID> & blk_group) const
+{
+  mooseAssert(_bnd_elem_range, "Boundary element range is not initialized");
+  const bool all_blocks =
+      std::find(blk_group.begin(), blk_group.end(), Moose::ANY_BLOCK_ID) != blk_group.end();
```

Comment:
```suggestion
  const bool all_blocks = blk_group.find(Moose::ANY_BLOCK_ID) != blk_group.end();
```
This is O(log(N)) as opposed to `std::find` which is O(N)